.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,62 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '34 2 * * 0'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: [ self-hosted, checkout-default-runners-github-runners ]
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
```

Comment:
This repo is for Android which uses Kotlin as a programming language. Also, we aim to integrate SonarQube on Frames soon. Is having CodeQL mandatory? CC @andra-lezza-cko. 
@jheng-hao-lin-cko @chintan-soni-cko @fabio-insolia-cko

Reply:
shall we merge this given that it's required?

Reply:
Yes we should merge this now as we now know we require both.

Reply:
Both SonarQube (SonarCloud for public repos) and CodeQL (GHAS) are required as they scan for different things - code quality and code security respectively. 
I just found this as well: https://codeql.github.com/docs/codeql-language-guides/codeql-for-java/
Not entirely sure why this workflow failed though as I can't see any errors and can't kick off a re-run..

Reply:
From the log it's something to do with the runner, update the runner to `ubuntu-latest` which has been applied to other Checkout Java SDK